### PR TITLE
TINY-6504: Enabled the `prefer-arrow/prefer-arrow-functions` ESLint rule for TinyMCE core

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6504

Description of Changes:
* Changes core to use arrow functions. This also means that all the old `namespace` declarations have been removed.
* Fixes the `StyleSheetLoader` not using a default export (all public TinyMCE APIs are supposed to use default exports).

Note: Core uses `this` a fair bit still unfortunately, so it'd probably be a good idea to properly review this (tests can be ignored). Some places that needed changes or needed linter rules disabled are:
-  EditorManager
-  NotificationManager
- DOMUtils

Places where namespace was removed:
- AddOnManager
- BookmarkManager
- DOMUtils
- RangeUtils
- SaxParser

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/test/ts/browser/EditorManagerTest.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ UnitTest.asynctest('browser.tinymce.core.EditorManagerTest', (success, failure)
       content_css: '/project/tinymce/js/tinymce/skins/content/default',
       language: langCode,
       language_url: langUrl,
-      init_instance_callback(_ed) {
+      init_instance_callback:  (_ed) => {
```
Comment (ltrouton) [inline modules/tinymce/src/core/test/ts/browser/EditorManagerTest.ts:90]: Looks like there is an additional space here?

File: modules/tinymce/src/core/main/ts/api/dom/RangeUtils.ts [hunk 2]
```diff
@@ -71,34 +71,32 @@ function RangeUtils(dom: DOMUtils): RangeUtils {
...
+ * Gets the caret range for the given x/y location.
+ *
+ * @static
+ * @method getCaretRangeFromPoint
+ * @param {Number} clientX X coordinate for range
+ * @param {Number} clientY Y coordinate for range
+ * @param {Document} doc Document that x/y are relative to
+ * @returns {Range} caret range
+ */
+RangeUtils.getCaretRangeFromPoint = CaretRangeFromPoint.fromPoint as (clientX: number, clientY: number, doc: Document) => Range;
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/api/dom/RangeUtils.ts:97]: Was wondering if the type assertions are required here?

File: modules/tinymce/src/core/main/ts/api/dom/BookmarkManager.ts [hunk 3]
```diff
@@ -66,18 +66,16 @@ function BookmarkManager(selection: EditorSelection): BookmarkManager {
...
-}
+/**
+ * Returns true/false if the specified node is a bookmark node or not.
+ *
+ * @static
+ * @method isBookmarkNode
+ * @param {DOMNode} node DOM Node to check if it's a bookmark node or not.
+ * @return {Boolean} true/false if the node is a bookmark node or not.
+ */
+BookmarkManager.isBookmarkNode = Bookmarks.isBookmarkNode as (node: Node) => boolean;
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/api/dom/BookmarkManager.ts:79]: Wondering if the type assertion is required here?

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/dom/RangeUtils.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/dom/RangeUtils.ts:97]: It was like this previously, so I'm hesitant to remove it. It shouldn't be needed though...

File: modules/tinymce/src/core/main/ts/api/dom/BookmarkManager.ts (hunk 3 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/dom/BookmarkManager.ts:79]: It was like this previously, so I'm hesitant to remove it. It shouldn't be needed though...

File: modules/tinymce/src/core/test/ts/browser/EditorManagerTest.ts (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/EditorManagerTest.ts:90]: Hmm... I wonder why the linter wasn't complaining about that :thinking: Either way, I blame my Mac and it's horrible keyboard that likes to do lots of double spaces :stuck_out_tongue:

Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/EditorManagerTest.ts:90]: Fixed. Will look into the potential linter issue separately (likely will be fixed by the eslint plugin upgrades).

File: modules/tinymce/src/core/main/ts/api/dom/RangeUtils.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/api/dom/RangeUtils.ts]: Removed the cast as discussed on slack

**Review by JamesToohey — APPROVED** [review]

**Pull request merged**